### DISCUSSION TITLE
i love to see you and SCH-lua

### DISCUSSION THREAD

**Discussion by binded-zz** [Deadlineem/Extras-Addon-for-YimMenu#43] [Ideas & Contributions]
i would love to see some of the sch-lua in your lua

seems you already have alot of stuff some what the same  but both dont have same things 
like ghost rider,  some of the npc stuff idk but both are great lua

thank you

**Comment by Deadlineem** [comment]
> i would love to see some of the sch-lua in your lua
> 
> seems you already have alot of stuff some what the same but both dont have same things like ghost rider, some of the npc stuff idk but both are great lua
> 
> thank you

We did add a few things from sch-lua in extras, If you could provide a detailed list of what you want ported from sch to extras, we can work on that!

**Comment by binded-zz** [comment]
i would love if you was able to pull all of it in to extras but in your layouts 
the global options  all of the player options 
protections  he does say people are free to use 


like pretty much grab most of it but convert to english and the layouts 
theres alot of it already in there or summarily functions

he has alot of npc controls 

a real lot of functions  but there a mess in the layout 

heres are some 


![sch](https://github.com/Deadlineem/Extras-Addon-for-YimMenu/assets/458540/783fb4ea-dc7b-456e-89fd-f009721f7772)
![sch2](https://github.com/Deadlineem/Extras-Addon-for-YimMenu/assets/458540/8ae7a9b1-ca05-401d-9d5b-316ae0d38186)

**Reply by Deadlineem** [reply]
Damn, ive never opened SCH-Lua in Yim.  Its so damn messy :( but there are indeed alot of features.  Id say more along the lines of picking the ones you use most out of the list of them.  @USBMenus @xesdoog what do you guys think?

**Reply by binded-zz** [reply]
also any way of maybe making the separate window movable  sch is so messy that on the player window its far far right 

![overlapped  far right](https://github.com/Deadlineem/Extras-Addon-for-YimMenu/assets/458540/f38032b5-9f34-4bc6-aa7c-7779f946b83d)

![float window](https://github.com/Deadlineem/Extras-Addon-for-YimMenu/assets/458540/e7b6acda-0dc5-47ea-8e9e-60c1f2bb8f34)
 unclicking the drops or griefs  makes sch float  as a debug window 

would love to see the mobile bussiness planning they have  like the show such computer
would be great on the business manger main page as there is nothing there 


bunch of things

**Reply by xesdoog** [reply]
> Damn, ive never opened SCH-Lua in Yim.  Its so damn messy :( but there are indeed alot of features.  Id say more along the lines of picking the ones you use most out of the list of them.  @USBMenus @xesdoog what do you guys think?

Sch is such a good programmer but UI isn't his main focus. We can definitely integrate some useful functions into Extras like instant sales in the business manager.
He also has developed a method to scan the memory for GTA Online's version to prevent outdated scripts from messing up the game. This is very important for any feature that sets globals like business restocking so we should add that method to features that are present in Extras.

**Comment by Deadlineem** [comment]
> > Damn, ive never opened SCH-Lua in Yim.  Its so damn messy :( but there are indeed alot of features.  Id say more along the lines of picking the ones you use most out of the list of them.  @USBMenus @xesdoog what do you guys think?
> 
> Sch is such a good programmer but UI isn't his main focus. We can definitely integrate some useful functions into Extras like instant sales in the business manager. He also has developed a method to scan the memory for GTA Online's version to prevent outdated scripts from messing up the game. This is very important for any feature that sets globals like business restocking so we should add that method to features that are present in Extras.

We should discuss a list of all the options we could use that would be beneficial for the community from sch-lua.  That option to scan the memory sounds like a nice fail-safe feature which would at the very least, allow us to have time to update the globals if it does not automatically do that itself.

**Reply by xesdoog** [reply]
It doesn't update it just checks the version and we can use it to not show any options if the version number is greater than the current one. Here's a very retarded [resupply script](https://github.com/xesdoog/YimResupplier/blob/main/YimResupplier.lua) that I made a while ago, take a look at the version check. We're basically going to do the same for Extras features that would break if the game updates.

**Comment by Deadlineem** [comment]
> It doesn't update it just checks the version and we can use it to not show any options if the version number is greater than the current one. Here's a very retarded [resupply script](https://github.com/xesdoog/YimResupplier/blob/main/YimResupplier.lua) that I made a while ago, take a look at the version check. We're basically going to do the same for Extras features that would break if the game updates.

I do recall seeing the version check in a few scripts including sch-lua.  That may be useful now that i think about it from that perspective, would help slim down on unnecessary bans.  However, i want to figure out ways to not need globals, or auto update them
